src/docfx/build/dependency/DependencyMapBuilder.cs

```diff
@@ -34,13 +35,63 @@ public void AddDependencyItem(Document from, Document to, DependencyType type)
 
         public DependencyMap Build()
         {
-            return new DependencyMap(_dependencyItems
-                .GroupBy(k => k.Source)
+            return new DependencyMap(Flatten());
+        }
+
+        private Dictionary<Document, HashSet<DependencyItem>> Flatten()
+        {
+            var result = new Dictionary<Document, HashSet<DependencyItem>>();
+            var graph = _dependencyItems
+                .GroupBy(k => k.From)
                 .ToDictionary(
                     k => k.Key,
-                    v => (from r in v.Distinct()
-                         orderby r.Dest.FilePath, r.Type
-                         select r).ToList()));
+                    v => (from r in v
+                          orderby r.To.FilePath, r.Type
```

Comment:
Just like `Dictionary`, `HashSet` also preserves insertion order, so the result is sorted and deterministic.

Reply:
yes, found that, so `order by` can be removed